src/lib/utils/hooks/SynapseAPI/download/useDownloadList.ts

```diff
@@ -0,0 +1,192 @@
+import {
+  ActionRequiredRequest,
+  AvailableFilesRequest,
+  AvailableFilter,
+  Sort,
+} from '../../../synapseTypes/DownloadListV2/QueryRequestDetails'
+import {
+  useInfiniteQuery,
+  UseInfiniteQueryOptions,
+  useMutation,
+  UseMutationOptions,
+  useQuery,
+  useQueryClient,
+  UseQueryOptions,
+} from 'react-query'
+import {
+  ActionRequiredResponse,
+  AvailableFilesResponse,
+  FilesStatisticsResponse,
+} from '../../../synapseTypes/DownloadListV2/QueryResponseDetails'
+import { SynapseClientError } from '../../../SynapseClientError'
+import { useSynapseContext } from '../../../SynapseContext'
+import { SynapseClient } from '../../../index'
+import { EntityHeader } from '../../../synapseTypes'
+import { AddBatchOfFilesToDownloadListResponse } from '../../../synapseTypes/DownloadListV2/AddBatchOfFilesToDownloadListResponse'
+
+export const downloadListQueryKeys = {
```

Comment:
Moved everything into one file with a shared base query key (so we can invalidate everything at once, e.g. when a file is added/removed).

This will help solve some existing cache issues in SWC, where files may be added to the download list without using react-query. We'll invalidate this base key when we fire the DownloadListUpdatedEvent.